bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -316,6 +324,27 @@ static boolean isProjectContainer(Container container) {
             tmp.add(JavaCore.newAccessRule(new Path("**"), IAccessRule.K_NON_ACCESSIBLE));
             return tmp.toArray(new IAccessRule[tmp.size()]);
         }
+        List<IAccessRule> tmp = null;
+        Packages exportPkgs = model.getExports();
+        ZipFile zip;
+        try {
+            zip = new ZipFile(c.getFile());
+            for (PackageRef exportPkg : exportPkgs.keySet()) {
+                String zipDir = exportPkg.getBinary();
+                ZipEntry ze = zip.getEntry(zipDir);
+                if (ze != null) {
+                    String pathStr = zipDir + "/*";
+                    if (tmp == null) {
+                        tmp = new LinkedList<IAccessRule>();
+                    }
+                    tmp.add(JavaCore.newAccessRule(new Path(pathStr), IAccessRule.K_ACCESSIBLE));
+                }
+            }
+        } catch (IOException e) {}
+        if (tmp != null && tmp.size() != 0) {
+            tmp.add(JavaCore.newAccessRule(new Path("**"), IAccessRule.K_NON_ACCESSIBLE));
+            return tmp.toArray(new IAccessRule[tmp.size()]);
```

Comment:
There is duplicate code here (add ** rule, convert to array). It would seem that it is better move that common code to the end of the method so all exit paths end here, no? That is, this method needs more refactoring for your change.